base/range_iterator.hpp

```diff
@@ -0,0 +1,101 @@
+#include "std/iterator.hpp"
+#include "std/type_traits.hpp"
+
+namespace base
+{
+
+// RagneItrator allows to write for loops as follows:
+// for (auto const i : range(N))
+//   ...
+// for (auto const i : range(M, N))
+//   ...
+// And initialize stl containers like this:
+// vector<int> mySequence(MakeRangeIterator(0), MakeRangeIterator(10));
+template <typename TCounter>
+struct RangeIterator
+{
+  explicit RangeIterator(TCounter const current) : m_current(current) { }
+
+  RangeIterator & operator++() { ++m_current; return *this; }
+  RangeIterator & operator--() { --m_current; return *this; }
```

Comment:
Please, follow our style guide. Postfix version looks good, but formatting of the prefix version should be fixed.